specification/_types/mapping/core.ts

```diff
@@ -99,54 +99,72 @@ export class KeywordProperty extends DocValuesPropertyBase {
   type: 'keyword'
 }
 
-export class NumberPropertyBase extends DocValuesPropertyBase {
+export class IPProperty extends DocValuesPropertyBase {
   index?: boolean
   ignore_malformed?: boolean
+  null_value?: string
+  on_script_error?: OnScriptError
+  script?: Script
+  /** [experimental] For internal use by Elastic only. Marks the field as a time series dimension. Defaults to false. */
```

Comment:
I think we can use the `@stability experimental` annotation on properties? We can add `@server_default false` too.